src/main/java/htsjdk/samtools/util/IntervalTree.java

```diff
@@ -492,7 +492,7 @@ public int getRelationship( final Node<V1> interval )
                 result = HAS_LESSER_PART;
             if ( mEnd > interval.getEnd() )
                 result |= HAS_GREATER_PART;
-            if ( mStart < interval.getEnd() && interval.getStart() < mEnd )
+            if ( mStart <= interval.getEnd() && interval.getStart() <= mEnd )
```

Comment:
thankfully all of these members are unused, which is why it went undetected.....the getLength function has no contract...so it's not clear what the intention was...but sure, I'll change it.